libcextract/SymbolExternalizer.cpp

```diff
@@ -71,142 +87,161 @@ using namespace llvm;
 /* IntervalTree.  */
 using namespace Intervals;
 
-// For debugging purposes.
-#if 0
-extern "C" void Debug_Range(const SourceRange &range)
-{
-  llvm::outs() << PrettyPrint::Get_Source_Text_Raw(range) << '\n';
-}
-
-#define DEBUG_RANGE(x) do { \
-  llvm::outs() << "line " << __LINE__ << "  "; \
-  Debug_Range(x); \
-  } while (0)
-#endif
-
-/** Define a Visitor just to update TypeOfType instances:
- *
- *  Kernel code sometime does things like this:
- *
- *   typeof(symbol_to_externalize) x;
- *
- * this bizarre constructs come from macros, which sometimes explodes because
- * clang-extract is unable to determine which part of it generates the Decl
- * in question.
- *
- * Now we have to update those typeofs, but there is no easy way of parsing Types
- * to get to the Expr.  But the RecursiveASTVisitor knows how to do it, so we
- * hack the class in order to setup a call to
- * FunctionUpdate::Update_References_To_Symbol and update those references.
- *
- */
 class TypeUpdaterVisitor : public RecursiveASTVisitor<TypeUpdaterVisitor>
 {
   public:
     /* Constructor. Should match the FunctionUpdate constructor so we can
        instantiate it in the Visitor.  */
-    TypeUpdaterVisitor(SymbolExternalizer &se, ValueDecl *new_decl,
-                       const std::string &old_decl_name, bool wrap)
-    : SE(se),
-      NewSymbolDecl(new_decl),
-      OldSymbolName(old_decl_name),
-      Wrap(wrap)
+    TypeUpdaterVisitor(SymbolExternalizer &se)
+    : SE(se)
     {}
 
   enum {
     VISITOR_CONTINUE = true,   // Return this for the AST transversal to continue;
     VISITOR_STOP     = false,  // Return this for the AST tranversal to stop completely;
   };
 
-  /* The updator method.  This will be called by the Visitor when traversing the
-     code.  */
-  bool VisitTypeOfExprType(TypeOfExprType *type)
+  bool VisitDeclaratorDecl(DeclaratorDecl *decl)
   {
-    /* Create a instance of our tested-in-battle FunctionUpdater...  */
-    SymbolExternalizer::FunctionUpdater fu(SE, NewSymbolDecl, OldSymbolName, Wrap);
+    SymbolUpdateStatus *sym = SE.getSymbolsUpdateStatus(decl->getName());
+    if (sym == nullptr)
+      return VISITOR_CONTINUE;
```

Comment:
Please explain why it skips this DeclaratorDecl if the SymbolUpdateStatus is `nullptr`. Something in the lines of "not in hash => not interesting symbol".